hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/filters/With.java

```diff
@@ -119,6 +122,39 @@ public static PropertyValues propertyValues(String name, Object... values) {
         return new PropertyValues(name, values);
     }
 
+    /**
+     * Checks whether there is data on the provided relative path. This filter is only really applicable
+     * on the {@link org.hawkular.inventory.api.model.DataEntity}. I.e. the relative path is always resolved
```

Comment:
well, basically the same thing as `RelationWith` does, i.e. narrowing the allowable filters only for the DataEntities. It's not typesafe, you can pass the `RelationWith.foo()` instance where normal filter is expected, but at least the api should look odd :]

```
inventory.tenants.getAll(RelationWith.name(foo)) X inventory.tenants.relationships.getAll(RelationWith.name(foo))
```

 Currently the data filters make sense only for resources, but it's on the `With` class that any entity filter can use.